packages/opentelemetry-core/src/common/attributes.ts

```diff
@@ -16,6 +16,19 @@
 
 import { diag, SpanAttributeValue, SpanAttributes } from '@opentelemetry/api';
 
+function serializeNonPrimitiveAttributeValue(
+  key: string,
+  value: unknown
+): SpanAttributeValue | undefined {
+  if (typeof value === 'object' && Array.isArray(value)) {
+    try {
+      return JSON.stringify(value);
+    } catch (error) {
+      diag.warn(`Invalid attribute value set for key: ${key}`);
```

Comment:
This warning doesn't really describe what happened. Most likely a circular reference or other unserializable thing